tests/integration/main.rs

```diff
@@ -0,0 +1,111 @@
+mod artichoke;
+
+use std::env;
+use std::path::PathBuf;
+use std::process::Command;
+
+use bstr::{BString, ByteSlice};
+use serde::ser::{Serialize, SerializeStruct, Serializer};
+
+struct CommandOutput {
+    call_args: Vec<String>,
+    status: i32,
+    stdout: BString,
+    stderr: BString,
+}
+
+impl CommandOutput {
+    fn new() -> Self {
+        Self {
+            call_args: vec![],
+            status: -1,
+            stdout: BString::from(""),
+            stderr: BString::from(""),
+        }
+    }
+
+    fn with_args(&mut self, call_args: &[&str]) -> &mut Self {
+        self.call_args
+            .append(&mut (*call_args).iter().map(|x| x.to_string()).collect());
+        self
+    }
+
+    fn with_command_output(&mut self, output: std::process::Output) -> &mut Self {
```

Comment:
This is mostly a style nit, I generally only use zero or one level of modules in item paths (except in macros).

```suggestion
    fn with_command_output(&mut self, output: Output) -> &mut Self {
```

Reply:
I love that you've passed this in by value so we can deconstruct it and take the outputs from it without cloning.